src/util/amount.rs

```diff
@@ -668,7 +668,11 @@ impl SignedAmount {
     ///
     /// Does not include the denomination.
     pub fn fmt_value_in(&self, f: &mut fmt::Write, denom: Denomination) -> fmt::Result {
-        fmt_satoshi_in(self.as_sat().abs() as u64, self.is_negative(), f, denom)
+        let sats = self.as_sat().checked_abs().map(|a: i64| a as u64).unwrap_or_else(|| {
+            // We could also hard code this into `9223372036854775808`
```

Comment:
Given that we only expect this to happen for a single value of `self.as_sat()`, should we not better assert it, return the fixed value and have the calculation commented as documentation?